components/performance-details/EditPerformanceModal.js

```diff
@@ -0,0 +1,284 @@
+  const [performanceLevelDropdownOptions, setPerformanceLevelDropdownOptions] = useState([]);
+  const [danceStyleDropdownOptions, setDanceStyleDropdownOptions] = useState([]);
+  const [danceSizeDropdownOptions, setDanceSizeDropdownOptions] = useState([]);
+
+  const getFilters = async () => {
+    setLoading(true);
+
+    const getSchools = async () => {
+      try {
+        const response = await axios({
+          method: 'GET',
+          url: '/api/schools/collect',
+        });
+        const schools = formatSchools(response.data);
+
+        // Modal dropdown options
+        setSchoolDropdownOptions(
+          formatDropdownOptions(schools, {
+            value: 'id',
+            label: 'schoolName',
+          })
+        );
+      } catch {
+        // Empty catch block
+      }
+    };
+
+    const getSettings = async () => {
+      const response = await axios({
+        method: 'GET',
+        url: '/api/settings/collect',
+      });
+      const settings = response.data;
+
+      const performanceLevelSettings = settings.filter(
+        setting => setting.type === 'COMPETITION_LEVEL'
+      );
+      const danceStyleSettings = settings.filter(setting => setting.type === 'STYLE');
+      const danceSizeSettings = settings.filter(setting => setting.type === 'DANCE_SIZE');
+
+      const formatOptionsFields = {
+        value: 'id',
+        label: 'value',
+      };
+
+      // Modal dropdown options
+      setPerformanceLevelDropdownOptions(
+        formatDropdownOptions(performanceLevelSettings, formatOptionsFields)
+      );
+      setDanceStyleDropdownOptions(formatDropdownOptions(danceStyleSettings, formatOptionsFields));
+      setDanceSizeDropdownOptions(formatDropdownOptions(danceSizeSettings, formatOptionsFields));
+    };
+
+    await getSchools();
+    await getSettings();
+
+    setLoading(false);
+  };
+
+  const clearFields = () => {
+    setDanceTitle('');
+    setDancersString('');
+    setChoreographersString('');
+    setSchool(null);
+    setCompetitionLevel(null);
+    setDanceStyle(null);
+    setDanceSize(null);
+  };
+
+  const onCancel = () => {
+    setOpen(false);
+    clearFields();
+  };
+
+  const updatePerformance = async () => {
+    const { id } = performance;
+    setLoading(true);
+
+    try {
+      await axios({
+        method: 'PUT',
+        url: '/api/performances/edit',
+        data: {
+          id,
+          name: danceTitle,
+          performers: dancersString.split(',').map(dancer => dancer.trim()),
+          choreographers: choreographersString
+            .split(',')
+            .map(choreographer => choreographer.trim()),
+          competitionLevel: competitionLevel.label,
+          competitionLevelID: competitionLevel.value,
+          danceSize: danceSize.label,
+          danceSizeID: danceSize.value,
+          danceStyle: danceStyle.label,
+          danceStyleID: danceStyle.value,
+          danceTitle,
```

Comment:
Careful with the name and danceTitle here.  Can you double to check to make sure we're actually cleaning this up on the frontend?

Reply:
We agreed on just using danceTitle.  I'll be pushing out the backend changes tonight for this.